packages/babel-plugin-transform-runtime/src/index.js

```diff
@@ -53,17 +54,17 @@ export default declare((api, options, dirname) => {
     rawVersion = corejs;
   }
 
-  const corejsVersion = rawVersion ? Number(rawVersion) : false;
+  const corejsVersion = rawVersion ? coerce(String(rawVersion)) : false;
 
-  if (![false, 2, 3].includes(corejsVersion)) {
+  if (corejsVersion !== false && !(2 <= corejsVersion.major <= 3)) {
```

Comment:
Since `core-js-pure` is a dependency of `@babel/runtime-corejs3` and missed in direct dependencies of users, unlike `preset-env`, it will not work without serious changes in the architecture.

Reply:
Indeed, this is a dead end for now.